Sources/App/WebView/WebViewController.swift

```diff
@@ -616,6 +654,43 @@ class WebViewController: UIViewController, WKNavigationDelegate, WKUIDelegate, U
         decisionHandler(.grant)
     }
 
+    @objc private func checkForBiometrics() {
+        if Current.settingsStore.biometricsRequired {
+            addBiometricOverlayProtection()
+            Current.authenticationService.delegate = self
+            Current.authenticationService.authenticate()
+        } else {
+            DispatchQueue.main.async { [weak self] in
+                self?.biometricOverlay.removeFromSuperview()
+            }
+        }
+    }
+
+    @objc private func protectAppIfNeeded() {
+        if Current.settingsStore.biometricsRequired {
+            addBiometricOverlayProtection()
+        } else {
+            DispatchQueue.main.async { [weak self] in
+                self?.biometricOverlay.removeFromSuperview()
+            }
+        }
+    }
+
+    private func addBiometricOverlayProtection() {
+        DispatchQueue.main.async { [weak self] in
+            guard let self else { return }
+            self.biometricOverlay.removeFromSuperview()
```

Comment:
make sure you set `accessibilityViewIsModal` and double check that VoiceOver can't get past this overlay

Reply:
I replaced the overlay by a full view controller